interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+        return representer.represent_dict(out)
+
+    def get_yaml(self, out):
+        out['name'] = self._name
+        out['thermo'] = self.thermo_model
+        out['elements'] = FlowList(self._el.split())
+
+        if len(self._sp) == 1 and self._sp[0][0] == 'species':
+            # all local species
+            out['species'] = self._sp[0][1]
+        else:
+            out['species'] = [BlockMap([(sp[0], sp[1])]) for sp in self._sp]
+
+        if 'skip_undeclared_elements' in self._options:
+            out['skip-undeclared-elements'] = True
+
+        # Convert reaction pattern matching to use of multiple reaction sections
+        for i in range(len(self._rx)):
+            spec = self._rx[i][1]
+            name = self._name + '-reactions'
+            if '*' in spec and name not in _reactions:
+                pattern = re.compile(spec.replace('*', '.*'))
+                misses = []
+                hits = []
+                for reaction in _reactions['reactions']:
+                    if pattern.match(reaction._id):
+                        hits.append(reaction)
+                    else:
+                        misses.append(reaction)
+                _reactions[name] = hits
+                _reactions['reactions'] = misses
+                self._rx[i] = [name, 'all']
+
+        if self._kin and self._rx:
+            out['kinetics'] = _newNames[self._kin]
+            if len(self._rx) == 1 and self._rx[0][0] == 'reactions':
+                out['reactions'] = self._rx[0][1]
+            elif all(r[1] == 'all' for r in self._rx):
+                out['reactions'] = FlowList(r[0] for r in self._rx)
+            else:
+                out['reactions'] = [BlockMap([(r[0], r[1])]) for r in self._rx]
+
+
+        if self._tr:
+            out['transport'] = _newNames[self._tr]
+
+        if self._comment:
+            out['note'] = self._comment
+
+        if self._initial:
+            out['state'] = self._initial
+
+
+class ideal_gas(phase):
+    """An ideal gas mixture."""
+    def __init__(self, name='', elements='', species='', note='',
+                 reactions='none', kinetics='GasKinetics', transport=None,
+                 initial_state=None, options=[]):
+        """
+        The parameters correspond to those of :class:`.phase`, with the
+        following modifications:
+
+        :param kinetics:
+            The kinetics model. Usually this field is omitted, in which case
+            kinetics model GasKinetics, appropriate for reactions in ideal gas
+            mixtures, is used.
+        :param transport:
+            The transport property model. One of the strings ``'none'``,
+            ``'multi'``, or ``'mix'``. Default: ``'none'``.
+        """
+
+        phase.__init__(self, name, 3, elements, species, note, reactions,
+                       initial_state, options)
+        self._kin = kinetics
+        self._tr = transport
+        self.thermo_model = 'ideal-gas'
+
+
+class stoichiometric_solid(phase):
```

Comment:
I may be mis-remembering, but I thought we were getting rid of `stoichiometric_solid` as a phase type? If not, the constructor arguments need documentation.

Reply:
I think it is on the list of things of probably-redundant `ThermoPhase` classes. But either way, I don't see the need to duplicate the description of the arguments when it already says that they are described for class `phase`.